src/app/modules/form/form.service.ts

```diff
@@ -174,3 +240,49 @@ export const getFormModelByResponseMode = (
       return EncryptedFormModel
   }
 }
+
+/**
+ * Checks if a form is accessed from within intranet and sets the property accordingly
+ * @param ip The ip of the request
+ * @param publicFormView The form to check
+ * @returns ok(PublicFormView) if the form is accessed from the internet
+ * @returns err(ApplicationError) if an error occured while checking if the ip of the request is from the intranet
+ */
+export const setIsIntranetFormAccess = (
+  ip: string,
+  publicFormView: IPublicFormView,
+): Result<SetRequired<IPublicFormView, 'isIntranetUser'>, ApplicationError> =>
+  IntranetFactory.isIntranetIp(ip).andThen((isIntranetIp) => {
+    const isIntranetUser =
+      isIntranetIp &&
+      [AuthType.SP, AuthType.CP, AuthType.MyInfo].includes(
+        publicFormView.form.authType,
+      )
```

Comment:
I don't think this logic is correct. what if the authType is NIL, but the IP is an intranet IP? then this function (and the subsequent HTTP response) returns `isIntranetUser: false`, even though it really is an intranet user.

right now the intranet code is used purely for SPCP/MyInfo, but that doesn't mean we can return false data when the authType is NIL. the same code may be reused for different purposes in future, and it will be very confusing for future developers if the backend returns `isIntranetUser: false` even when the IP is an intranet IP.

Reply:
good catch!